frontend/src/components/auth/teacher-signup/steps/TeacherSignUpOne.tsx

```diff
@@ -73,31 +79,51 @@ const TeacherSignupOne = ({
   };
 
   const validateFields = (): boolean => {
+    let isValid = true;
+
     if (!watch("firstName") || !!errors.firstName) {
       setFirstNameError(true);
-      return false;
+      isValid = false;
+    } else {
+      setFirstNameError(false);
     }
 
     if (!watch("lastName") || !!errors.lastName) {
       setLastNameError(true);
-      return false;
+      isValid = false;
+    } else {
+      setLastNameError(false);
     }
 
-    if (!watch("email") || !!errors.email) {
+    const emailValue = watch("email");
+    if (!emailValue || !!errors.email) {
       setEmailError(true);
-      return false;
+      isValid = false;
+    } else {
+      setEmailError(false);
+    }
+
+    if (!watch("email") || (emailValue && !emailRegex.test(emailValue))) {
```

Comment:
```suggestion
    if (!emailRegex.test(emailValue)) {
```
we don't need to check for the empty case again since this was done above